src/coreclr/src/jit/lower.cpp

```diff
@@ -4488,10 +4495,20 @@ GenTree* Lowering::LowerNonvirtPinvokeCall(GenTreeCall* call)
                 break;
 
             case IAT_PPVALUE:
+                // ToDo:  Expanding an IAT_PPVALUE here, loses the opportunity
+                // to Hoist/CSE the first indirection as it is an invariant load
```

Comment:
Are you going to open an issue for this?

Reply:
It would speed up PInkvoke calls that occur inside of a loop in a crossgen-ed assembly.
I will investigate how often this condition actually occurs.

Reply:
The following two methods in System.Private.Corelib have PInvoke calls inside of a loop:

```
Top method regressions (bytes):
          12 ( 3.74% of base) : System.Private.CoreLib.dasm - System.GC:GetTotalMemory(bool):long
           4 ( 0.35% of base) : System.Private.CoreLib.dasm - System.RuntimeType:GetInterfaceMap(System.Type):System.Reflection.InterfaceMapping:this
```